scripts/setup-k8s-master.sh.in

```diff
@@ -92,29 +107,15 @@ kubectl apply -f {{StorageClassUrl}}
 # Set up the network policy blocking the AWS metadata endpoint from the default namespace.
 kubectl apply -f {{NetworkPolicyUrl}}
 
-INSTANCE_ID=$(ec2metadata --instance-id)
-# Add this machine (master) to the load balancer for external access
-aws elb register-instances-with-load-balancer \
-  --load-balancer-name {{LoadBalancerName}} \
-  --instances ${INSTANCE_ID} \
-  --region {{Region}}
-
 # Use kubeadm's kubeconfig command to grab a client-cert-authenticated
 # kubeconfig file for administrative access to the cluster.
 KUBECONFIG_OUTPUT=/home/ubuntu/kubeconfig
 
-# TODO(@chuckha): --apiserver-advertise-address is resolved to an IP address.
-# We don't want this to happen because we need to use the ELB load balancer as the
-# api server address.
-# Instead we set the server by hand.
 kubeadm alpha phase kubeconfig user \
   --client-name admin \
-  --apiserver-advertise-address "${LB_DNS}" \
+  --apiserver-advertise-address "${LB_IPV4}" \
```

Comment:
This *should* use the value of `ControlPlaneEndpoint` here. Unless that is not the case, then we should be able to drop the --apiserver-advertise-address flag here.

Reply:
Fixed, thanks!

Reply:
After chatting with @chuckha, seems this issue is related to https://github.com/kubernetes/kubernetes/issues/18174 and https://github.com/kubernetes/kubernetes/issues/19161. Seems we might still need to overwrite it.